semanticcpg/src/main/scala/io/shiftleft/semanticcpg/accesspath/AccessPath.scala

```diff
@@ -49,9 +51,10 @@ object AccessPath {
       val nElements = el.elements.length - 1
       while (nElements - i > -1) {
         el.elements(nElements - i) match {
-          case (AddressOf | VariablePointerShift | _: PointerShift) => i += 1
+          case (AddressOf | VariablePointerShift | _: PointerShift) =>
           case _                                                    => return i
         }
+        i += 1
```

Comment:
Why?

Reply:
Do we need this unaesthetic while loop here?

Reply:
This specific change is purely for readability -- it does not change the control flow graph, and if the scala compiler does its job then it should not even change the bytecode.

Regarding whether we need the loop at all -- this is hot code, and I see no formulation that is both more obvious in terms of semantics and more obvious in terms of performance (i.e. in terms of what we hope for the style of machine code).

Reply:
To me the old version was the better one.